addons/point_of_sale/models/pos_session.py

```diff
@@ -335,7 +335,7 @@ def _create_picking_at_end_of_session(self):
         else:
             session_destination_id = picking_type.default_location_dest_id.id
 
-        for order in self.order_ids:
+        for order in self.order_ids.filtered(lambda o: o.to_ship is not True):
```

Comment:
```suggestion
        for order in self.order_ids.filtered(lambda o: not o.to_ship):
```
or maybe best to use `filtered_domain`